Source/Codecov/Coverage/Tool/Coverage.cs

```diff
@@ -67,7 +76,7 @@ private IEnumerable<ReportFile> LoadReportFile()
                         ? expanded
                         : Enumerable.Empty<string>())
                 .Select(x => new ReportFile(x, File.ReadAllText(x))).ToArray();
-            if (report?.Any() != true)
+            if (report.Length == 0)
```

Comment:
Don't remove the question mark here, at this point the report **CAN** still be null.

Reply:
Any input or output of LINQ methods can not be `null`. If `Enumerable.Empty` is supplied, `Select` returns `Enumerable.Empty` and `ToArray` returns `Array.Empty`; if `null` is supplied you'll just get an `ArgumentNullException`. You can see this clearly with `#nullable enable`.

Reply:
Yeah, I missed the first question mark in the call chain.

Reply:
Not exactly. As per CommandLineParser docs:

> The IEnumerables are initialized to empty arrays rather than null.

IEnumerables in .NET are usually non-nullable.

Reply:
> Not exactly. As per CommandLineParser docs:

Yes, their docs say that. However, I have had cases in the past where it would still be set to null. So having a null check here is still required.

> IEnumerables in .NET are usually non-nullable.

Typically yes, but there is nothing stopping it from being null (especially when the program/library isn't marked with the nullable attribute).